tests/run/i1846.scala

```diff
@@ -0,0 +1,22 @@
+object Test {
```

Comment:
I think is it better to use `assert` instead of a check file for simple test cases like this one. Also one more think you could test:
```scala
val foo = 42 match { case x @ { 42 } => x }
assert(foo == 42)
```